### ISSUE TITLE
Wrong link to rust book in note

### ISSUE THREAD

**Issue by hellow554** [rust-lang/rust#54555]
```rust
fn main() {
    let a: str;
}
```

([Playground](https://play.rust-lang.org/?gist=eb9ebc74259b45a88d8b12ece4bcc023&version=stable&mode=debug&edition=2015))

Errors:

```
   Compiling playground v0.0.1 (file:///playground)
error[E0277]: the size for values of type `str` cannot be known at compilation time
 --> src/main.rs:2:9
  |
2 |     let a: str;
  |         ^ doesn't have a size known at compile-time
  |
  = help: the trait `std::marker::Sized` is not implemented for `str`
  = note: to learn more, visit <https://doc.rust-lang.org/book/second-edition/ch19-04-advanced-types.html#dynamically-sized-types-and-sized>
  = note: all local variables must have a statically known size

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
error: Could not compile `playground`.

To learn more, run the command again with --verbose.

```

The note contains a wrong link

    [...]/ch19-04-advanced-types.html#dynamically-sized-types-and-sized

vs 

    [...]/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait

As easy as this fix could be, I grepped for the error and found ~120 occurences, most of them in the tests of course. 
While I want to solve this, I ask myself, if the best strategy is a simple `sed` or is there a better and not so error-prone approach? If one link changes in the future, all of this has to be done again. I don't think that that is a good solution :(

**Comment by steveklabnik** [comment]
The second edition is frozen, and so that link will no longer change.

**Comment by Havvy** [comment]
For this specific case, just fixing it with `sed` is probably a good idea. For future proofing though, perhaps `linkcheck` could be amended to fix that? Though that's probably non-trivial.

**Comment by ehuss** [comment]
I'm a little confused.  This should have been fixed in #53082, unless you are referring to an older version of rustc?

**Comment by Havvy** [comment]
If it's already fixed, there's no reason keeping the issue open.
